validator/src/bootstrap.rs

```diff
@@ -407,20 +407,25 @@ pub fn attempt_download_genesis_and_snapshot(
         .map_err(|err| format!("Failed to get RPC node slot: {err}"))?;
     info!("RPC node root slot: {}", rpc_client_slot);
 
-    download_snapshots(
-        full_snapshot_archives_dir,
-        incremental_snapshot_archives_dir,
-        validator_config,
-        bootstrap_config,
-        use_progress_bar,
-        maximum_local_snapshot_age,
-        start_progress,
-        minimal_snapshot_download_speed,
-        maximum_snapshot_download_abort,
-        download_abort_count,
-        snapshot_hash,
-        rpc_contact_info,
-    )?;
+    let snapshot_from_file =
+        validator_config.snapshot_config.snapshot_from == Some(SnapshotFrom::File);
+
+    if !snapshot_from_file {
+        download_snapshots(
```

Comment:
Might be out of scope for this PR. If we fail our load from files, we try from the existing snapshot archives, but we don't download new ones. Is that correct?

Reply:
Maybe we should consider checking the current snapshot slot. If it's too old, then we download and use the downloaded archive regardless of `snapshot_from_file`?